internal/stores/hostdb.go

```diff
@@ -151,23 +252,72 @@ func (db *SQLStore) Hosts(notSince time.Time, max int) ([]hostdb.Host, error) {
 	return hosts, err
 }
 
+func hostByPubKey(tx *gorm.DB, hostKey consensus.PublicKey) (dbHost, error) {
+	var h dbHost
+	err := tx.Where("public_key", gobEncode(hostKey)).
+		Take(&h).Error
+	return h, err
+}
+
 // RecordInteraction records an interaction with a host. If the host is not in
 // the store, a new entry is created for it.
-func (db *SQLStore) RecordInteraction(hostKey consensus.PublicKey, hi hostdb.Interaction) error {
+func (db *SQLStore) RecordHostInteractions(hostKey consensus.PublicKey, interactions []hostdb.Interaction) error {
+	dbInteractions := make([]dbInteraction, len(interactions))
+	var successful, failed float64
+
+	var scanQuery string
+	var scanArgs []interface{}
+	for i, interaction := range interactions {
+		dbInteractions[i] = dbInteraction{
+			Result:    interaction.Result,
+			Success:   interaction.Success,
+			Timestamp: interaction.Timestamp.UTC(),
+			Type:      interaction.Type,
+		}
+		if interaction.Success {
+			successful++
+		} else {
+			failed++
+		}
+		// Handle scan edge case.
+		if interaction.Type == hostdb.InteractionTypeScan {
+			timeToAddTo := "uptime"
+			if !interaction.Success {
+				timeToAddTo = "downtime"
+			}
+			var sr hostdb.ScanResult
+			if interaction.Success {
+				if err := json.Unmarshal(interaction.Result, &sr); err != nil {
+					return err
+				}
+			}
+			scanQuery += fmt.Sprintf(`UPDATE hosts
```

Comment:
couldn't we simply discard all interactions except for the last 2, 
then update the total scans to add the total amount of interactions,
and then update `previous_scan_success` and `last_scan_success` 
according to the last two interactions?

Reply:
I think this method is missing code like:
if 1 interaction -> do this
if 2 interactions -> do this

I think you wrote the code for when it's 1 interaction, but if we have way more we probably want to optimise the hell out of this and squash  it down to 2 statements:

`UPDATE hosts SET total_scan ... `
`UPDATE hosts SET successful_interactions...`

and personally I would rewrite it to have a single statement, but I can see why you didn't